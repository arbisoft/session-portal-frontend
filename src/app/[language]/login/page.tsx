import type { Metadata } from "next";

import LoginPage from "@/features/LoginPage";
import { getServerTranslation } from "@/services/i18n";

type Props = {
  params: { language: string };
};

export async function generateMetadata({ params }: Props): Promise<Metadata> {
  const { t } = await getServerTranslation(params.language, "login");

  return {
    title: t("title"),
  };
}

export default LoginPage;
