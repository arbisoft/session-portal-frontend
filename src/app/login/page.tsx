import type { Metadata } from "next";

import LoginPage from "@/features/LoginPage";

export async function generateMetadata(): Promise<Metadata> {
  return {
    title: "Sessions Portal",
  };
}

export default LoginPage;
