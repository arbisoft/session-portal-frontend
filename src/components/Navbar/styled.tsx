import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import InputBase, { inputBaseClasses } from "@mui/material/InputBase";
import { alpha, css, styled } from "@mui/material/styles";
import { typographyClasses } from "@mui/material/Typography";

import { pxToRem } from "@/utils/styleUtils";

export const StyledAppBar = styled(AppBar, { name: "StyledAppBar" })(() => {
  return css`
    position: fixed;
  `;
});

export const Logo = styled("div", { name: "Logo" })(({ theme }) => {
  return css`
    align-items: center;
    cursor: pointer;
    display: flex;
    width: 240px;

    ${theme.breakpoints.down("lg")} {
      width: unset;
    }

    svg {
      color: red;
      margin-right: 8px;
    }

    .${typographyClasses.h6} {
      color: ${theme.palette.common.white};
      font-size: ${pxToRem(16)};
      font-style: normal;
      font-weight: 700;
      line-height: normal;
      text-decoration: none;
    }
  `;
});

export const Search = styled("form", { name: "Search" })(({ theme }) => {
  return css`
    align-items: center;
    background-color: ${theme.palette.mode === "dark" ? alpha(theme.palette.common.white, 0.15) : "#e3e3e3"};
    border-radius: ${theme.shape.borderRadius * 2}px;
    display: flex;
    justify-content: space-between;
    margin-left: 0;
    width: 450px;
    overflow: hidden;

    border: ${theme.palette.mode === "light" ? "1px solid #A5A5A5" : "unset"};

    &:hover {
      background-color: ${alpha(theme.palette.common.white, 0.25)};
    }

    ${theme.breakpoints.down("lg")} {
      width: 200px;
    }
  `;
});

export const SearchIconWrapper = styled(Button, { name: "SearchIconWrapper" })(({ theme }) => {
  return css`
    align-items: center;
    background-color: ${theme.palette.mode === "dark" ? alpha(theme.palette.common.white, 0.25) : "#cccccc"};
    border-bottom-right-radius: ${theme.shape.borderRadius * 2}px;
    border-radius: 0;
    border-top-right-radius: ${theme.shape.borderRadius * 2}px;
    border: none;
    color: inherit;
    display: flex;
    height: 100%;
    justify-content: center;
    padding: ${theme.spacing(0, 2)};
  `;
});

export const CancelIconWrapper = styled("div", { name: "CancelIconWrapper" })(({ theme }) => {
  return css`
    padding-right: ${theme.spacing(1)};
    padding-top: ${theme.spacing(1)};
    cursor: pointer;
  `;
});

export const StyledInputBase = styled(InputBase, { name: "StyledInputBase" })(
  ({ theme }) => css`
    color: inherit;
    width: 100%;
    & .${inputBaseClasses.input} {
      padding: ${theme.spacing(1, 1, 1, 1)};
      padding-right: calc(1em + ${theme.spacing(1)});
      background-color: transparent;
    }
  `
);

export const NavbarRightArea = styled(Box, { name: "NavbarRightArea" })(
  ({ theme }) => css`
    display: flex;
    flex-grow: 0;
    gap: ${theme.spacing(2)};
    justify-content: flex-end;
    width: 240px;

    ${theme.breakpoints.down("lg")} {
      width: unset;
    }
  `
);
