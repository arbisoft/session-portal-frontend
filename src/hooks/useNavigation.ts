import { useRouter } from "next/navigation";

export const ALLOWED_PAGES = {
  get home(): string {
    return "/";
  },
  get login(): string {
    return "/login";
  },
  get videos(): string {
    return "/videos";
  },
  get searchResult(): string {
    return `${this.videos}/results`;
  },
  videoDetail(id: string) {
    return `/videos/${id}`;
  },
  get uploadVideo(): string {
    return "/upload-video";
  },
} as const;

type Page = keyof typeof ALLOWED_PAGES;

const useNavigation = () => {
  const router = useRouter();

  /**
   * Generate a dynamic URL based on the page and optional params.
   * @param page - The target page key from ALLOWED_PAGES.
   * @param params - Query parameters or required dynamic ID (for `videoDetail`).
   * @returns Fully constructed localized URL.
   */
  const getPageUrl = (page: Page, params?: Record<string, string | number>) => {
    let pageUrl = ALLOWED_PAGES[page];

    if (typeof pageUrl === "function" && params?.id) {
      pageUrl = pageUrl(String(params.id));
      delete params.id;
    }

    const queryString =
      params && Object.keys(params).length ? `?${new URLSearchParams(params as Record<string, string>).toString()}` : "";

    return `${pageUrl}${queryString}`;
  };

  /**
   * Navigate to a new page dynamically.
   * @param page - The target page.
   * @param params - Query parameters or required `id` for `videoDetail`.
   */
  const navigateTo = (page: Page, params?: Record<string, string | number>) => {
    const url = getPageUrl(page, params);
    router.push(url, { scroll: true });
  };

  return { getPageUrl, navigateTo };
};

export default useNavigation;
