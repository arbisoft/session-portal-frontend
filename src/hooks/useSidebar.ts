import { useEffect, useState } from "react";

const useSidebar = () => {
  const [sidebarItems, setSidebarItems] = useState<string[]>([]);

  useEffect(() => {
    const onGetSidebarItems = async () => {
      //   TODO: Remove static code and get items from backend
      setSidebarItems(["All", "All-Hands Meetings", "AMA Sessions", "Product Showcase", "Nurture Sessions", "Security Session"]);
    };
    onGetSidebarItems();
  }, []);

  return { sidebarItems };
};
export default useSidebar;
