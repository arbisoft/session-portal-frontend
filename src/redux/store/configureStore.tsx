"use client";

import { configureStore, combineReducers } from "@reduxjs/toolkit";
import type { Reducer } from "redux";
import { persistStore, REHYDRATE, FLUSH, PAUSE, PERSIST, PURGE, REGISTER, persistReducer } from "redux-persist";
import createWebStorage from "redux-persist/lib/storage/createWebStorage";
import createFilter from "redux-persist-transform-filter";

import { baseApi, REDUCER_PATH } from "@/redux/baseApi";
import loginReducer, { LoginState } from "@/redux/login/slice";

export const createNoopStorage = () => {
  return {
    getItem(_key: string) {
      return Promise.resolve(_key);
    },
    setItem(_key: string, value: string) {
      return Promise.resolve(value);
    },
    removeItem(_key: string) {
      return Promise.resolve(_key);
    },
  };
};

export interface ReducersState {
  [REDUCER_PATH]: (typeof baseApi)["reducer"] extends Reducer<infer T> ? T : never;
  login: LoginState;
}

const storage = typeof window !== "undefined" ? createWebStorage("local") : createNoopStorage();

const loginFilter = createFilter("login", ["session"], ["session"]);

const persistConfig = {
  key: "session-portal",
  storage,
  transforms: [loginFilter],
  whitelist: ["login"],
};

const rootReducer = combineReducers({
  login: loginReducer,
  [baseApi?.reducerPath]: baseApi?.reducer,
});
const persistedReducer = persistReducer<ReturnType<typeof rootReducer>>(persistConfig, rootReducer);
const store = configureStore({
  devTools: process.env.NODE_ENV !== "production",
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }).concat([baseApi.middleware]),
  reducer: persistedReducer,
  // @ts-ignore
  preloadedState: global.preloadedState,
});

const persistor = persistStore(store);

export { store, persistor };
