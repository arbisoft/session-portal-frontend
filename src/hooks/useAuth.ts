import { useEffect } from "react";

import { usePathname } from "next/navigation";
import { useSelector } from "react-redux";

import { selectAccessToken } from "@/redux/login/selectors";

import useNavigation from "./useNavigation";

const useAuth = () => {
  const pathname = usePathname();
  const token = useSelector(selectAccessToken);
  const { navigateTo, getPageUrl } = useNavigation();

  useEffect(() => {
    const isRootOrLoginPage = getPageUrl("home").includes(pathname) || pathname === getPageUrl("login");

    if (token && isRootOrLoginPage) {
      navigateTo("videos");
    } else if (!token) {
      navigateTo("login");
    }
  }, [token, pathname]);
};

export default useAuth;
