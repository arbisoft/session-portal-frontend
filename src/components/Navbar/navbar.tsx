import React, { useEffect, useState } from "react";

import CancelIcon from "@mui/icons-material/Cancel";
import MenuIcon from "@mui/icons-material/Menu";
import SearchIcon from "@mui/icons-material/Search";
import Avatar from "@mui/material/Avatar";
import Box from "@mui/material/Box";
import Container from "@mui/material/Container";
import IconButton from "@mui/material/IconButton";
import Menu from "@mui/material/Menu";
import MenuItem from "@mui/material/MenuItem";
import { useTheme } from "@mui/material/styles";
import Toolbar from "@mui/material/Toolbar";
import Tooltip from "@mui/material/Tooltip";
import Typography from "@mui/material/Typography";
import Image from "next/image";
import { useSearchParams } from "next/navigation";
import { useDispatch, useSelector } from "react-redux";

import { useFeatureFlags } from "@/hooks/useFeatureFlags";
import useNavigation from "@/hooks/useNavigation";
import { selectUserInfo } from "@/redux/login/selectors";
import { loginActions } from "@/redux/login/slice";
import { persistor } from "@/redux/store/configureStore";

import ThemeToggle from "../ThemeToggle";

import { CancelIconWrapper, Logo, NavbarRightArea, Search, SearchIconWrapper, StyledAppBar, StyledInputBase } from "./styled";

function Navbar({ onDrawerToggle, shouldShowDrawer = false }: { onDrawerToggle?: VoidFunction; shouldShowDrawer?: boolean }) {
  const dispatch = useDispatch();
  const { navigateTo } = useNavigation();
  const searchParams = useSearchParams();
  const { isFeatureEnabled } = useFeatureFlags();
  const theme = useTheme();

  const isDarkModeVisible = isFeatureEnabled("darkModeSwitcher");
  const isUploadVideoVisible = isFeatureEnabled("uploadVideo");

  const userInfo = useSelector(selectUserInfo);

  const [searchQuery, setSearchQuery] = useState("");
  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);

  const search = searchParams?.get("search");

  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleCloseUserMenu = () => {
    setAnchorElUser(null);
  };

  const handleLogout = async () => {
    dispatch(loginActions.logout());
    await persistor.purge();
    persistor.persist();
    handleCloseUserMenu();
  };

  const handleSearch = (searchEvent: React.FormEvent) => {
    searchEvent.preventDefault();
    if (searchQuery.length > 0) {
      navigateTo("searchResult", { search: searchQuery });
    }
  };

  const handleClearSearch = () => {
    setSearchQuery("");
    navigateTo("videos");
  };

  useEffect(() => {
    setSearchQuery(search ?? "");
  }, [search]);

  return (
    <StyledAppBar data-testid="navbar">
      <Container maxWidth={false}>
        <Toolbar disableGutters>
          {shouldShowDrawer && (
            <IconButton
              color="inherit"
              data-testid="open-drawer"
              sx={{ marginRight: theme.spacing() }}
              aria-label="open drawer"
              edge="start"
              onClick={onDrawerToggle}
            >
              <MenuIcon />
            </IconButton>
          )}
          <Logo data-testid="navbar-logo" onClick={() => navigateTo("videos")}>
            <Image src={"/assets/images/apple-icon.png"} width={24} height={24} alt="Arbisoft Icon" data-testid="arbisoftLogo" />
            <Typography variant="h6" noWrap sx={{ display: { xs: "none", md: "flex" }, marginLeft: theme.spacing(1) }}>
              Arbisoft Sessions Portal
            </Typography>
          </Logo>

          <Box sx={{ flexGrow: 1, display: "flex", justifyContent: "center" }}>
            <Search onSubmit={handleSearch}>
              <StyledInputBase
                value={searchQuery}
                onChange={(inputEvent: React.ChangeEvent<HTMLInputElement>) => setSearchQuery(inputEvent.target.value)}
                placeholder={"Search..."}
                inputProps={{ "aria-label": "search", "data-testid": "search-query" }}
              />

              {searchQuery.length > 0 && (
                <CancelIconWrapper data-testid="cancelIcon" onClick={handleClearSearch}>
                  <CancelIcon />
                </CancelIconWrapper>
              )}

              <SearchIconWrapper type="submit">
                <SearchIcon data-testid="SearchIcon" />
              </SearchIconWrapper>
            </Search>
          </Box>
          <NavbarRightArea>
            {isDarkModeVisible && <ThemeToggle />}
            <Tooltip title="Open settings">
              <IconButton size="large" data-testid="avatar-btn" onClick={handleOpenUserMenu} sx={{ p: 0 }}>
                <Avatar variant="rounded" alt={userInfo.full_name ?? ""} src={userInfo.avatar ?? ""} />
              </IconButton>
            </Tooltip>
            <Menu
              data-testid="profile-menu"
              anchorEl={anchorElUser}
              anchorOrigin={{
                vertical: "bottom",
                horizontal: "right",
              }}
              keepMounted
              transformOrigin={{
                vertical: "top",
                horizontal: "right",
              }}
              open={Boolean(anchorElUser)}
              onClose={handleCloseUserMenu}
            >
              {isUploadVideoVisible && (
                <MenuItem onClick={() => navigateTo("uploadVideo")}>
                  <Typography>Upload a video</Typography>
                </MenuItem>
              )}
              <MenuItem data-testid="Logout" onClick={handleLogout}>
                <Typography>Logout</Typography>
              </MenuItem>
            </Menu>
          </NavbarRightArea>
        </Toolbar>
      </Container>
    </StyledAppBar>
  );
}

export default Navbar;
