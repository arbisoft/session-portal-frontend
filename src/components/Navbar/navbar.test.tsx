import { ReactElement } from "react";

import { configureStore } from "@reduxjs/toolkit";
import { useSearchParams } from "next/navigation";
import { Provider, useDispatch, useSelector } from "react-redux";

import { useFeatureFlags } from "@/hooks/useFeatureFlags";
import { fireEvent, screen, waitFor, render, RenderOptions, act } from "@/jest/utils/testUtils";
import { selectUserInfo } from "@/redux/login/selectors";
import { loginActions } from "@/redux/login/slice";
import { persistor } from "@/redux/store/configureStore";

import ThemeProvider from "../theme/theme-provider";

import Navbar from "./navbar";

jest.mock("next/navigation", () => ({
  ...jest.requireActual("next/navigation"),
  useSearchParams: jest.fn(),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock("@/redux/store/configureStore", () => ({
  persistor: {
    purge: jest.fn(),
    persist: jest.fn(),
  },
}));

const mockNavigateTo = jest.fn();
jest.mock("@/hooks/useNavigation", () => ({
  __esModule: true,
  default: () => ({
    navigateTo: mockNavigateTo,
  }),
}));

const mockStore = configureStore({
  reducer: {
    login: () => ({
      userInfo: {
        full_name: "John Doe",
        avatar: "https://example.com/avatar.jpg",
      },
    }),
  },
});

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={mockStore}>
        <ThemeProvider>{children}</ThemeProvider>
      </Provider>
    ),
    ...options,
  });

describe("Navbar Component", () => {
  const mockDispatch = jest.fn();
  const mockUserInfo = {
    full_name: "John Doe",
    avatar: "https://example.com/avatar.jpg",
  };

  beforeEach(() => {
    (useSelector as unknown as jest.Mock).mockImplementation((selector) =>
      selector === selectUserInfo ? mockUserInfo : undefined
    );

    (useDispatch as unknown as jest.Mock).mockReturnValue(mockDispatch);
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn().mockReturnValue("test search"),
    });

    (useFeatureFlags as jest.Mock).mockReturnValue({
      isFeatureEnabled: (feature: string) => feature === "test",
    });

    jest.clearAllMocks();
  });

  test("should render Navbar without crashing", async () => {
    customRender(<Navbar />);
    await waitFor(() => {
      expect(screen.getByTestId("navbar")).toBeInTheDocument();
    });
  });

  test("should render arbisoft logo and text", async () => {
    customRender(<Navbar />);
    await waitFor(() => {
      expect(screen.getByText("Arbisoft Sessions Portal")).toBeInTheDocument();
      expect(screen.getByTestId("arbisoftLogo")).toBeInTheDocument();
    });
  });

  test("should render avatar button", async () => {
    customRender(<Navbar />);
    await waitFor(() => {
      expect(screen.getByTestId("avatar-btn")).toBeInTheDocument();
    });
  });

  test("should open user menu when avatar is clicked", async () => {
    customRender(<Navbar />);
    fireEvent.click(screen.getByTestId("avatar-btn"));
    await waitFor(() => {
      expect(screen.getByText("Logout")).toBeInTheDocument();
    });
  });

  test("should close user menu when menu item is clicked", async () => {
    customRender(<Navbar />);
    fireEvent.click(screen.getByTestId("avatar-btn"));
    fireEvent.click(screen.getByTestId("Logout"));

    await waitFor(() => {
      expect(screen.getByTestId("profile-menu")).toHaveAttribute("aria-hidden", "true");
    });
  });

  test("should focus on search input when clicked", () => {
    customRender(<Navbar />);
    const searchInput = screen.getByTestId("search-query");
    act(() => {
      searchInput.focus();
    });
    expect(searchInput).toHaveFocus();
  });

  test("should navigate on search submit", () => {
    customRender(<Navbar />);
    const searchInput = screen.getByPlaceholderText("Search...");
    fireEvent.change(searchInput, { target: { value: "some test search" } });
    fireEvent.submit(searchInput.closest("form")!);
    expect(mockNavigateTo).toHaveBeenCalledWith("searchResult", { search: "some test search" });
  });

  test("should display tooltip on avatar button hover", async () => {
    customRender(<Navbar />);

    await act(async () => {
      fireEvent.mouseOver(screen.getByTestId("avatar-btn"));
    });

    expect(await screen.findByText("Open settings")).toBeInTheDocument();
  });

  test("should clear search input when cancel icon is clicked", () => {
    customRender(<Navbar />);
    const searchInput = screen.getByTestId("search-query");
    fireEvent.change(searchInput, { target: { value: "Test Query" } });
    expect(searchInput).toHaveValue("Test Query");
    fireEvent.click(screen.getByTestId("cancelIcon"));
    expect(searchInput).toHaveValue("");
  });

  test("should call logout and purge persistor when logout is clicked", async () => {
    customRender(<Navbar />);
    fireEvent.click(screen.getByTestId("avatar-btn"));
    fireEvent.click(screen.getByText("Logout"));

    expect(mockDispatch).toHaveBeenCalledWith(loginActions.logout());
    expect(persistor.purge).toHaveBeenCalled();
    await waitFor(() => expect(persistor.persist).toHaveBeenCalled());
  });

  test("should render ThemeToggle in the user menu", () => {
    (useFeatureFlags as jest.Mock).mockReturnValue({
      isFeatureEnabled: (feature: string) => feature === "darkModeSwitcher",
    });
    customRender(<Navbar />);
    fireEvent.click(screen.getByRole("dark-mode-button"));
    expect(screen.getByRole("dark-mode-button")).toBeInTheDocument();
  });

  test("should navigate to the videos page when logo is clicked", () => {
    customRender(<Navbar />);
    fireEvent.click(screen.getByTestId("navbar-logo"));
    expect(mockNavigateTo).toHaveBeenCalledWith("videos");
  });

  test("should match snapshot", () => {
    const { asFragment } = customRender(<Navbar />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("hides Upload Video button when feature flag is disabled", () => {
    (useFeatureFlags as jest.Mock).mockReturnValue({
      isFeatureEnabled: () => false,
    });

    customRender(<Navbar />);

    expect(screen.queryByText("Upload Video")).not.toBeInTheDocument();
  });

  it("shows Upload Video button when feature flag is enabled", () => {
    (useFeatureFlags as jest.Mock).mockReturnValue({
      isFeatureEnabled: (feature: string) => feature === "uploadVideo",
    });

    customRender(<Navbar />);

    expect(screen.getByText("Upload a video")).toBeInTheDocument();
  });
});
