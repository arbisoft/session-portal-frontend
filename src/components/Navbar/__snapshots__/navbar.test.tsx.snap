// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-b4hu5c-MuiPaper-root-MuiAppBar-root-StyledAppBar-root"
    data-testid="navbar"
    style="--Paper-shadow: var(--mui-shadows-4); --Paper-overlay: var(--mui-overlays-4);"
  >
    <div
      class="MuiContainer-root css-12f197v-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-1u9o2lj-MuiToolbar-root"
      >
        <div
          class="css-1vbiuhu-Logo-root"
          data-testid="navbar-logo"
        >
          <img
            alt="Arbisoft Icon"
            data-nimg="1"
            data-testid="arbisoftLogo"
            decoding="async"
            height="24"
            loading="lazy"
            src="/_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=48&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=32&q=75 1x, /_next/image?url=%2Fassets%2Fimages%2Fapple-icon.png&w=48&q=75 2x"
            style="color: transparent;"
            width="24"
          />
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1k6p2lw-MuiTypography-root"
          >
            Arbisoft Sessions Portal
          </h6>
        </div>
        <div
          class="MuiBox-root css-1rqqp05"
        >
          <form
            class="css-15t2c4j-Search-root"
          >
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary css-1kssppa-MuiInputBase-root-StyledInputBase-root"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input css-1943rd7-MuiInputBase-input"
                data-testid="search-query"
                placeholder="Search..."
                type="text"
                value="test search"
              />
            </div>
            <div
              class="css-8brctg-CancelIconWrapper-root"
              data-testid="cancelIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
                data-testid="CancelIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2m5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12z"
                />
              </svg>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1cdkzx6-MuiButtonBase-root-MuiButton-root-SearchIconWrapper-root"
              tabindex="0"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-4uxqju-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
            </button>
          </form>
        </div>
        <div
          class="MuiBox-root css-si7qf5-NavbarRightArea-root"
        >
          <button
            aria-label="Open settings"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-9nlk2f-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            data-testid="avatar-btn"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-rounded css-11kq5jj-MuiAvatar-root"
            >
              <img
                alt="John Doe"
                class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
                src="https://example.com/avatar.jpg"
              />
            </div>
          </button>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
